packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -316,9 +317,15 @@ export async function syncEngineListen(
       queryToTarget(query)
     );
 
-    const status = syncEngineImpl.sharedClientState.addLocalQueryTarget(
-      targetData.targetId
-    );
+    // PORTING NOTE: When the query is listening to cache only, the target ID
```

Comment:
Maybe:

When the query is listening to cache only, we skip sending it over to Watch by not registering it in shared client state, and directly calculate initial snapshots and subsequent updates from cache.